activestorage/lib/active_storage/service/azure_storage_service.rb

```diff
@@ -51,16 +51,32 @@ def delete(key)
       end
     end
 
+    def delete_prefixed(prefix)
+      instrument :delete_all, prefix: prefix do
+        marker = nil
+
+        loop do
```

Comment:
Could this potentially block forever? Or does `list_blobs` time out? Does it raise an exception when it does?

Reply:
We can provide the `:timeout` option to `list_blobs` if we wish, but it returns a limited number of blobs for each call (5,000 by default), so I don’t see why it’d block indefinitely.

Reply:
👍, the `loop do` just had me a bit worried 😊